lib/private/Repair/RepairUnencryptedSize.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * @copyright Copyright (c) 2023, Nextcloud GmbH.
+ *
+ * @author Simon Spannagel <simonspa@kth.se>
+ *
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program. If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+namespace OC\Repair;
+
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\IRepairStep;
+
+class RepairUnencryptedSize implements IRepairStep {
+	/** @var IDBConnection */
+	protected $connection;
+
+	public function __construct(IDBConnection $connection) {
+		$this->connection = $connection;
+	}
+
+	public function getName() {
+		return 'Repair file size of unencrypted files';
+	}
+
+	/**
+	 * Fix unencrypted file size
+	 */
+	public function run(IOutput $output) {
+		$update = $this->connection->getQueryBuilder();
+		$update->update('filecache')
+			->set('unencrypted_size', "0")
+			->where($update->expr()->eq('encrypted', "0", IQueryBuilder::PARAM_INT))
+			->andWhere($update->expr()->neq('unencrypted_size', "0", IQueryBuilder::PARAM_INT));
```

Comment:
There is no index on neither of those columns, so we may want to add at least to the unencrypted size.

I think the most safe way to use such an index then would be:

```suggestion
			->andWhere($update->expr()->gt('unencrypted_size', "0", IQueryBuilder::PARAM_INT));
			->andWhere($update->expr()->eq('encrypted', "0", IQueryBuilder::PARAM_INT))
```

Reply:
For the reasoning:
- neq may not use the index
- the order of where statements sometimes also matters to decide if the index is applied

Reply:
EXPLAIN of the current query does a full table scan:

```

explain update oc_filecache set unencrypted_size = 0 where encrypted = 0 and unencrypted_size != 0;
+------+-------------+--------------+-------+---------------+---------+---------+------+------+-------------+
| id   | select_type | table        | type  | possible_keys | key     | key_len | ref  | rows | Extra       |
+------+-------------+--------------+-------+---------------+---------+---------+------+------+-------------+
|    1 | SIMPLE      | oc_filecache | index | NULL          | PRIMARY | 8       | NULL | 8479 | Using where |
+------+-------------+--------------+-------+---------------+---------+---------+------+------+-------------+
```

Reply:
I don't see how the index is applied after reversal of arguments and using `>`:

```
explain update oc_filecache set unencrypted_size = 0 where unencrypted_size > 0 and encrypted = 0;
+------+-------------+--------------+-------+---------------+---------+---------+------+------+-------------+
| id   | select_type | table        | type  | possible_keys | key     | key_len | ref  | rows | Extra       |
+------+-------------+--------------+-------+---------------+---------+---------+------+------+-------------+
|    1 | SIMPLE      | oc_filecache | index | NULL          | PRIMARY | 8       | NULL | 490  | Using where |
+------+-------------+--------------+-------+---------------+---------+---------+------+------+-------------+
```

Reply:
I implemented the suggested change - but adding an index to either of these columns would have to done separately, and probably run manually, right? How would that work together with an automatic repair step?